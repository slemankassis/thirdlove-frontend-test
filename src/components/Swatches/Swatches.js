import React from 'react';
import RoundSelector from '../../thirdy-part-components/RoundPicker';

class Swatches extends React.Component {
  render() {
    console.log(this.props);

    return (
      <RoundSelector />
    );
  }
}

export default Swatches;
