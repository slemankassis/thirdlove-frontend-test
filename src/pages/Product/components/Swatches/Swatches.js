import React from 'react';
import RoundPicker from '../../../../thirdy-part-components/RoundPicker';

class Swatches extends React.Component {
  render() {
    return (
      this.props.colors.map((color) => <RoundPicker key={color} value={color} />)
    );
  }
}

export default Swatches;
