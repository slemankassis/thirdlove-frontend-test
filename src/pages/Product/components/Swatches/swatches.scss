$colors: (
  #ffefe5,
  #dab5a1,
  #d49d89,
  #6a3a33,
  #524346
);
