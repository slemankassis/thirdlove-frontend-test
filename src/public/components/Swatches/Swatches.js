const React = require('react');
const RoundSelector = require('./../RoundPicker');

class Swatches extends React.Component {
  render() {
    return (
      <RoundSelector />
    );
  }
}

module.exports = Swatches;
