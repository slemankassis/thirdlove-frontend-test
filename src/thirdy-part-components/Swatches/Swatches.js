import React from 'react';
import PropTypes from 'prop-types';

class Swatches extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);

    this.state = {
      selected: this.props.selected,
    };
  }

  handleChange(value) {
    this.setState(() => ({
      selected: value,
    }));
    this.props.onChange(value);
  }

  render() {
    const { options } = this.props;
    return (
      options.map((value) => (
        <div key={value}>
          <input type="radio" id={value} value={value} onChange={() => this.handleChange(value)} name="swatches" checked={value === this.state.selected} />
          <label htmlFor={value}>{value}</label>
        </div>
      ))
    );
  }
}

Swatches.propTypes = {
  selected: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func,
};

Swatches.defaultProps = {
  onChange: null,
};

export default Swatches;
