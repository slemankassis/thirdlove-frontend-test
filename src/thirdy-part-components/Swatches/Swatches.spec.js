import React from 'react';
import Swatches from './Swatches';

describe('Swatches snaps', () => {
  it('renders snapshot with basicProps', () => {
    const basicProps = {
      selected: '__SELECTED__',
      options: ['__OPTION_1__', '__OPTION_2__'],
    };
    const component = shallow(<Swatches {...basicProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('renders snapshot with fullProps', () => {
    const fullProps = {
      selected: '__SELECTED__',
      options: ['__OPTION_1__', '__OPTION_2__'],
      onChange: jest.fn(),
    };

    const component = shallow(<Swatches {...fullProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
