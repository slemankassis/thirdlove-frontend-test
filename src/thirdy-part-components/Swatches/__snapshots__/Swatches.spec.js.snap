// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swatches renders snapshot with basicProps 1`] = `
"<div>
  <input type='radio' id='__OPTION_1__' value='__OPTION_1__' onChange={[Function: onChange]} name='swatches' checked={false} />
  <label htmlFor='__OPTION_1__'>
    __OPTION_1__
  </label>
</div>


<div>
  <input type='radio' id='__OPTION_2__' value='__OPTION_2__' onChange={[Function: onChange]} name='swatches' checked={false} />
  <label htmlFor='__OPTION_2__'>
    __OPTION_2__
  </label>
</div>"
`;

exports[`Swatches renders snapshot with fullProps 1`] = `
"<div>
  <input type='radio' id='__OPTION_1__' value='__OPTION_1__' onChange={[Function: onChange]} name='swatches' checked={false} />
  <label htmlFor='__OPTION_1__'>
    __OPTION_1__
  </label>
</div>


<div>
  <input type='radio' id='__OPTION_2__' value='__OPTION_2__' onChange={[Function: onChange]} name='swatches' checked={false} />
  <label htmlFor='__OPTION_2__'>
    __OPTION_2__
  </label>
</div>"
`;
